utils/S3UtilsMongoClient.js

```diff
@@ -150,6 +190,39 @@ class S3UtilsMongoClient extends MongoClientInterface {
                 results.location[rep.site] = size;
             }
         });
+
+        // count in both dataStoreName and cold location if object is restored
+        if (entry.value.archive
+            && entry.value.archive.restoreCompletedAt <= Date.now()
+            && entry.value.archive.restoreWillExpireAt >= Date.now()) {
+            entry.value.location.forEach(location => {
```

Comment:
I think if we are exactly at the expiration time, it should be excluded from the check, as the object is considered expired

```suggestion
        if (entry.value.archive
            && entry.value.archive.restoreCompletedAt <= Date.now()
            && entry.value.archive.restoreWillExpireAt > Date.now()) {
            entry.value.location.forEach(location => {
```